### ISSUE TITLE
flex array / struct hack static init not supported by suncc

### ISSUE THREAD

**Issue by nigoroll** [varnishcache/varnish-cache#2137]
as discussed on IRC, flex arrays are supported, but apparently not for static init
http://docs.oracle.com/cd/E19205-01/820-7598/6nirjuni7/index.html

```
  CC       varnishtest-vtc_varnish.o
"vtc_h2_dectbl.h", line 17: too many struct/union initializers (E_TOO_MANY_STRUCT_UNION_INIT)
"vtc_h2_dectbl.h", line 25: too many struct/union initializers (E_TOO_MANY_STRUCT_UNION_INIT)
```

using a generator, it should be easy to use something like

```
struct stbl {
        int msk;
        struct ssym *syms;
};

static struct ssym byte3_pref0110_syms[2] = {
        /* idx   0 */ {1, 199, NULL},
        /* idx   1 */ {1, 207, NULL},
};

static struct stbl byte3_pref0110 = {
        1, byte3_pref0110_syms
};
```

**Comment by nigoroll** [comment]
btw, should the length of the arrays be stored somewhere maybe, just as a precaution, of at least be terminated by some null element? I have not looked at the code _at all_, just asking

**Comment by gquintard** [comment]
The length is known using mask (it's the log2 of the array length). I'll deal with it.

**Comment by nigoroll** [comment]
@gquintard close or do you want to solve this differently?

**Comment by gquintard** [comment]
Closing

**Comment by hermunn** [comment]
Backport review: Nothing commited, nothing to backport.

**Comment by hermunn** [comment]
Correction: This was fixed in bc2ba0a07791d1ae7, and since this is H2, no backport is needed.
